lib/fluent/config/element.rb

```diff
@@ -112,6 +117,28 @@ def to_s(nest = 0)
         out
       end
 
+      def to_masked_element
+        element = Element.new(@name, @arg, {}, @elements, @unused)
+        each_pair { |k, v|
+          element[k] = secret_param?(k) ? 'xxxxxx' : v
+        }
+        element
+      end
+
+      def secret_param?(key)
+        return false if @corresponding_proxies.empty?
+
+        param_key = key.to_sym
+        @corresponding_proxies.each { |proxy|
+          block, opts = proxy.params[param_key]
+          if opts
```

Comment:
Should it be `opts && opts[:secret]`?
If opts has other option's key, this method returns opts[:secret] if it's not specified.